gcloud/gcloud.go

```diff
@@ -0,0 +1,147 @@
+// Copyright © 2019 Percona, LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gcloud
+
+import (
+	"fmt"
+	"math/rand"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+)
+
+const charset = "abcdefghijklmnopqrstuvwxyz" +
+	"ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
+
+var seededRand = rand.New(rand.NewSource(time.Now().UnixNano()))
+
+// Gcloud env structure for GKE
+type Gcloud struct {
+	envName     string
+	project     string
+	zone        string
+	cluster     string
+	keyfile     string
+	namespace   string
+	execCommand string
+}
+
+// New gcloud object
+func New(name string, project string, zone string, cluster string, keyfile string, namespace string) (*Gcloud, error) {
+	perconaHome := os.Getenv("HOME") + "/.percona"
+	gcloudobj := &Gcloud{}
+
+	gcloudobj.execCommand = k8sExecDefault
+	if _, err := exec.LookPath(gcloudobj.execCommand); err != nil {
+		gcloudobj.execCommand = k8sExecCustom
+		if _, err := exec.LookPath(gcloudobj.execCommand); err != nil {
+
+			fmt.Println("Can't find kubectl executable. Installing it according to https://kubernetes.io/docs/tasks/tools/install-kubectl/#install-kubectl-on-linux.")
+			_, err := exec.Command("curl", "-LO", "https://storage.googleapis.com/kubernetes-release/release/v1.15.0/bin/linux/amd64/kubectl").CombinedOutput()
+			if err != nil {
+				return nil, fmt.Errorf("binary download has been failed. %v", err)
+			}
+
+			_, err = exec.Command("chmod", "+x", "./kubectl").CombinedOutput()
```

Comment:
cloud we set permissions via pure golang?

Reply:
done